# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/libcubeb/src/cubeb_wasapi.cpp
# Commit: 9794b0a3bffc
# Full Hash: 9794b0a3bffc474951b33527b738488717caffbf
# Author: Alex Chronopoulos <achronop@gmail.com>
# Date: 2019-12-12 21:50:40
# Regressor Bug: 1603384
# File Overlap Count: 1
# Description:
#   Bug 1603384 - Update cubeb from upstream to aa63601. r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D56872
# ==============================================================================

diff -r 7d77d727cd1e -r 9794b0a3bffc media/libcubeb/src/cubeb_wasapi.cpp
--- a/media/libcubeb/src/cubeb_wasapi.cpp	Thu Dec 12 17:46:46 2019 +0200
+++ b/media/libcubeb/src/cubeb_wasapi.cpp	Thu Dec 12 15:24:23 2019 +0000
@@ -1921,14 +1921,14 @@
 
     /* Get a client. We will get all other interfaces we need from
      * this pointer. */
-    // hr = device->Activate(__uuidof(IAudioClient3),
-    //                       CLSCTX_INPROC_SERVER,
-    //                       NULL, audio_client.receive_vpp());
-    // if (hr == E_NOINTERFACE) {
+    hr = device->Activate(__uuidof(IAudioClient3),
+                          CLSCTX_INPROC_SERVER,
+                          NULL, audio_client.receive_vpp());
+    if (hr == E_NOINTERFACE) {
       hr = device->Activate(__uuidof(IAudioClient),
                             CLSCTX_INPROC_SERVER,
                             NULL, audio_client.receive_vpp());
-    //}
+    }
 
     if (FAILED(hr)) {
       LOG("Could not activate the device to get an audio"
@@ -1994,16 +1994,16 @@
     flags |= AUDCLNT_STREAMFLAGS_EVENTCALLBACK;
   }
 
-  // if (initialize_iaudioclient3(audio_client, stm, mix_format, flags, direction)) {
-  //   LOG("Initialized with IAudioClient3");
-  // } else {
+  if (initialize_iaudioclient3(audio_client, stm, mix_format, flags, direction)) {
+    LOG("Initialized with IAudioClient3");
+  } else {
     hr = audio_client->Initialize(AUDCLNT_SHAREMODE_SHARED,
                                   flags,
                                   frames_to_hns(stm, stm->latency),
                                   0,
                                   mix_format.get(),
                                   NULL);
-  // }
+  }
   if (FAILED(hr)) {
     LOG("Unable to initialize audio client for %s: %lx.", DIRECTION_NAME, hr);
     return CUBEB_ERROR;
